File: twosum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> v1;
        
        for(int i=0;i<nums.size();i++)
        {
            v1.push_back({nums[i],i});
        }
        sort(v1.begin(),v1.end());
        int n=nums.size();
        int i=0,j=n-1;
        vector<int> ans;
        while(i<j)
        {
            int sum=v1[i].first+v1[j].first;
            if(sum==target)
            {
                return {v1[i].second,v1[j].second};
            }
            else if(sum>target)
            {
                j--;
            }
            else
            {
                i++;
            }
        }
        
        return {};
            
    }
};

File: twosum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        n = len(nums)
        for i in range(n):
            for j in range(n):
                if i != j and target == nums[i] + nums[j]:
                    return [i, j]

    def twoSum1(self, nums: list[int], target: int) -> list[int]:
        n = len(nums)
        for i in range(n):
            if target - nums[i] in nums:
                ind = nums.index(target - nums[i])
                if ind != i:
                    return [i, ind]

    def twoSum2(self, nums: list[int], target: int) -> list[int]:
        lookup = {}
        for index, num in enumerate(nums):
            if target - num in lookup:
                return[lookup[target - num], index]
            lookup[num] = index
        return []


sol = Solution()
print(sol.twoSum1(list(range(1, 10001)), 19999))
